authors/apps/authentication/tests/social_authentication_tests.py

```diff
@@ -0,0 +1,75 @@
+"""This module runs tests for user login process via Facebook, Twitter or Google"""
+import json
+import os
+from django.test import TestCase
+from rest_framework.test import APIClient
+from rest_framework.reverse import reverse as api_reverse
+from rest_framework import status
+
+
+class UserTestCase(TestCase):
+    """This class contains tests for the user login process through Twiiter, Facebook or Google"""
+    def setUp(self):
+        """This function defines variables to be used throughout the class"""
+        self.client = APIClient()
+        self.google_key = os.getenv('GOOGLE_KEY')
+        self.google_secret = os.getenv('GOOGLE_SECRET')
+        self.facebook_key = os.getenv('FACEBOOK_KEY')
+        self.facebook_secret = os.getenv('FACEBOOK_SECRET')
+        self.twitter_key = os.getenv('TWITTER_KEY')
+        self.twitter_secret = os.getenv('TWITTER_SECRET')
+ 
+    def test_google_login_no_tokens(self):
+        """this method test that a user cannot login via Google if there's no access_key and access_key_secret"""
+
+        data = {
+            "provider": "google"}
+        response = self.client.post('api/socialAuth', data=data)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_google_login(self):
+        """this method test that a user can login through google"""
+
+        data = {
+            "provider": "google", 
+            "access_key": self.google_key, 
+            "access_key_secret": self.google_secret}
+        response = self.client.post('api/socialAuth', data=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_twitter_login_no_tokens(self):
+        """this method test that a user cannot login via Twitter if there's no access_key and access_key_secret"""
+
+        data = {
+            "provider": "twitter"}
+        response = self.client.post('api/socialAuth', data=data)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_twitter_login(self):
+        """this method test that a user can login through twitter"""
+
+        data = {
+            "provider": "twitter", 
+            "access_key": self.twitter_key, 
+            "access_key_secret": self.twitter_secret}
+        response = self.client.post('api/socialAuth', data=data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_facebook_login_no_tokens(self):
+        """this method test that a user cannot login via Facebook if there's no access_key and access_key_secret"""
```

Comment:
Good use of docstrings. However you did not follow pep8. 
https://www.python.org/dev/peps/pep-0008/#maximum-line-length

Reply:
Sorry, I had Pylinted the file but somewhere in the rebase or squash or sth I lost the changes.